src/components/ListItem.jsx

```diff
@@ -1,9 +1,49 @@
 import './ListItem.css';
+import { useEffect, useCallback } from 'react';
+import { updateItem } from '../api/firebase';
+
+const millisecondsIn24hrs = 86400000;
+const currentDate = new Date();
+const currentTime = currentDate.getTime();
+
+export function ListItem({ listToken, item }) {
+	let calcDate = item.dateLastPurchased
+		? item.dateLastPurchased.seconds * 1000
+		: null;
+
+	useEffect(() => {
+		let timer = currentTime - calcDate;
+		if (item.isChecked && timer > millisecondsIn24hrs) {
+			updateItem(listToken, item, {
+				isChecked: !item.isChecked,
+			});
+		}
+	}, [item, listToken, calcDate]);
```

Comment:
the dependency array should be empty so that this part of code runs only on the first render